vue-frontend/src/components/Authentication.vue

```diff
@@ -36,7 +36,7 @@
         </v-card-media>
         <v-card-actions>
           <v-spacer />
-          <v-btn flat class="orange--text" :loading="pending" type="submit">Sign In</v-btn>
+          <v-btn flat class="orange--text" :rules="[students.username!=''&& students.password!='' ? true : false]" :loading="studentLoading" type="submit">Sign In</v-btn>
```

Comment:
What happened to the loading state?

Reply:
`studentLoading` and `educatorLoading` are two functions that could return the loading state. Agreed with your later comment that it might be better to just add a dimension onto the authentication state and avoid this confusing logic.